fix(build): empty `outDir` before build

underfin opened the pull request.

underfin committed:
fix(build): remove `outDir` before build

aleclarson reviewed (changes requested):
It should only empty the `outDir` when `config.write` is true.

Also, what if the build fails? Is it better to leave the previous contents intact then?

underfin committed:
fix(build): add `write` condition

underfin commented:
> Also, what if the build fails? Is it better to leave the previous contents intact then?

If user want to build a project represent to them want to remove previous content. This is not atomic and don't consider fallback case.

aleclarson committed:
fix(build): always render the index.html

If `build` or `ssrBuild` are called programmatically with `write: false`, the index.html output still needs to exist in the returned array of `BuildResult` objects.
